src/dynimage.rs

```diff
@@ -629,27 +629,24 @@ impl DynamicImage {
     pub fn grayscale(&self) -> DynamicImage {
         match *self {
             DynamicImage::ImageLuma8(ref p) => DynamicImage::ImageLuma8(p.clone()),
-            DynamicImage::ImageLumaA8(ref p) => {
-                DynamicImage::ImageLumaA8(imageops::grayscale_alpha(p))
-            }
-            DynamicImage::ImageRgb8(ref p) => DynamicImage::ImageLuma8(imageops::grayscale(p)),
-            DynamicImage::ImageRgba8(ref p) => {
-                DynamicImage::ImageLumaA8(imageops::grayscale_alpha(p))
-            }
+
+            DynamicImage::ImageLumaA8(ref p) => DynamicImage::ImageLumaA8(p.clone()),
+
+            DynamicImage::ImageRgb8(_) => dynamic_map!(*self, |ref p| DynamicImage::ImageLuma8(p.convert())), 
+
+            DynamicImage::ImageRgba8(_) => dynamic_map!(*self, |ref p| DynamicImage::ImageLumaA8(p.convert())),
```

Comment:
I'm not sure why the dynamic_map is useful here. The dynamic_map! macro just expands to a match statement exactly like the one we're currently in:
https://github.com/image-rs/image/blob/37d3de296cf256c17e14de71d212849cbe026542/src/dynimage.rs#L107-L118

Reply:
Also, just saw that I didn't remove the markdown comment to accept the license agreement, I edited the main comment to include it

Reply:
I would expect that this would be at least as fast as your dynamic_map! version:
```
DynamicImage::ImageRgba8(p) => DynamicImage::ImageLumaA8(p.convert()),
```